### PR TITLE
[Cortx-dev]EOS-13952:Old elasticsearch data cleanup cron script modification

### PR DESCRIPTION
**Pull request by udayan-y21**

### CODE REVIEW

File: csm/conf/csm_cleanup.py [hunk 1]
```diff
@@ -79,38 +79,41 @@ def remove_old_indexes(es, arg_d, arg_n, arg_e):
...
-    es = esCleanup(const.CSM_CLEANUP_LOG_FILE, const.CSM_LOG_PATH)
-    remove_old_indexes(es, args.d, args.n, args.e)
 
-def process_auditlogs(args):
+def process_es_cleanup(args):
     # Pass arguments to worker function
     # remove data older than given number of days
     es = esCleanup(const.CSM_CLEANUP_LOG_FILE, const.CSM_LOG_PATH)
-    es.remove_old_data_from_indexes(args.d, args.n, args.i, args.f)
+    es.remove_old_data_from_indexes(args.d, args.n, ["csmauditlog"], "timestamp")
```
Comment (kupranay) [inline csm/conf/csm_cleanup.py:87]: extract it as a function... add logs in debug

File: csm/conf/csm_cleanup.py [hunk 2]
```diff
@@ -79,38 +79,41 @@ def remove_old_indexes(es, arg_d, arg_n, arg_e):
...
+    remove_old_indexes(es, args.d, args.n, args.e)
+
+    cmd = "du -BM /var/log/elasticsearch/"
+    sp = SimpleProcess(cmd)
+    res = sp.run()
+    res = res[0]
+    res = int(res.decode("utf-8").split('\t')[0].split('M')[0])
+    es_db_capping = (int(args.s) * int(args.e)) / 100
+
+    while int(args.p) > int(args.c):
```
Comment (kupranay) [inline csm/conf/csm_cleanup.py:98]: while loop  args.p needs to be updated

**Review by kupranay — CHANGES_REQUESTED** [review]

File: csm/conf/csm_cleanup.py (hunk 1 above)
Comment (udayan-y21) [reply csm/conf/csm_cleanup.py]: Comment addresed. Please review

File: csm/conf/csm_cleanup.py (hunk 2 above)
Comment (udayan-y21) [reply csm/conf/csm_cleanup.py]: Comment addresed. Please review

**Review by navalkp — APPROVED** [review]

File: csm/conf/csm_cleanup.py [hunk 3]
```diff
@@ -79,38 +86,70 @@ def remove_old_indexes(es, arg_d, arg_n, arg_e):
         else:
             Log.debug("Nothing to remove")
 
-def process_stats(args):
-    # Pass arguments to worker function
-    es = esCleanup(const.CSM_CLEANUP_LOG_FILE, const.CSM_LOG_PATH)
-    remove_old_indexes(es, args.d, args.n, args.e)
+def clean_indexes(es, no_of_days, host_port):
+    for index, field in index_field_map.items():
```
Comment (ajaysrivas) [inline csm/conf/csm_cleanup.py:90]: rename field to timestamp_field.

File: csm/conf/csm_cleanup.py [hunk 4]
```diff
@@ -79,38 +86,70 @@ def remove_old_indexes(es, arg_d, arg_n, arg_e):
...
 
-def process_auditlogs(args):
+def execute_cmd(cmd=""):
+    sp_es = SimpleProcess(cmd)
+    Log.debug(f"Running {cmd}")
+    res = sp_es.run()
+    Log.debug(f"Resulted: {res}")
+    return res
+
+def get_du_data():
```
Comment (ajaysrivas) [inline csm/conf/csm_cleanup.py:102]: get_du_data should get_dir_usage(dir)
get_df_data should get_fs_uasge(fs)

Both of these function should be in common code.

**Review by ajaysrivas — CHANGES_REQUESTED** [review]

File: csm/conf/csm_cleanup.py (hunk 3 above)
Comment (udayan-y21) [reply csm/conf/csm_cleanup.py]: Comments resolved.Please review.

File: csm/conf/csm_cleanup.py (hunk 4 above)
Comment (udayan-y21) [reply csm/conf/csm_cleanup.py]: Comments resolved.Please review.

**Review by ajaysrivas — APPROVED** [review]

File: csm/common/storage_usage.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        """
+
+        cmd = f"df -B{unit} {fs}"
+        return StorageInfo.execute_cmd(cmd)
+
+    @staticmethod
+    def execute_cmd(cmd=""):
+        sp_es = SimpleProcess(cmd)
+        Log.debug(f"Running {cmd}")
+        res = sp_es.run()
```
Comment (prathameshrodi) [inline csm/common/storage_usage.py:55]: This will Return a Tuple of `output`, `err` and `rc`
log them correctly as  o/p, err, and rc below

**Review by prathameshrodi — CHANGES_REQUESTED** [review]

File: csm/common/storage_usage.py (hunk 5 above)
Comment (udayan-y21) [reply csm/common/storage_usage.py:55]: Comments addressed. Please reviewa nad approve

**Review by prathameshrodi — APPROVED** [review]

**Review by kupranay — APPROVED** [review]

**Comment by udayan-y21** [comment]
@kupranay Added the comment in https://jts.seagate.com/browse/EOS-13280 for codacy issue conflict in Error D202 and D205.
Please merge this PR

**Pull request merged**